core/interop/src/loaders/merge.rs

```diff
@@ -0,0 +1,68 @@
+#![allow(clippy::module_name_repetitions)]
+//! A [`ModuleLoader`] that "merges" two module loaders into one. It will
+//! try to resolve using the first loader, and if it fails, it will try the
+//! second.
+//!
+//! Any errors from the first loader will be ignored, and the second loader
+//! will be used.
+
+use std::rc::Rc;
+
+use boa_engine::module::{ModuleLoader, Referrer};
+use boa_engine::{Context, JsResult, JsString, Module};
+
+/// A [`ModuleLoader`] that "merges" two module loaders into one. It will
+/// try to resolve using the first loader, and if it fails, it will try the
+/// second.
+#[derive(Debug)]
+pub struct MergeModuleLoader<First, Second> {
+    first: First,
+    // When resolving modules we need to clone the second loader.
+    second: Rc<Second>,
```

Comment:
Thought: Hmmm, I don't really like this because we already wrap `MergeModuleLoader` in an `Rc`. Maybe it's time to commit to the `Rc` wrapper and change the receiver of the `ModuleLoader` methods to `Rc<Self>`.

Reply:
Not actionable from your side, I'm just documenting my thoughts.